docs/source/developers/components_exports.rst

```diff
@@ -0,0 +1,483 @@
+================================
+
+Components are, simply put, subapplications. They have their own templates,
+satic assets, route handlers, database migrations, and so on. Each of these
+things are called 'component members'. Some members may require load ordering
+(e.g., plugins may depend on other plugins, some routes may require to be
+registered after others or before others). We call these ordering requirements
+'component member dependencies'.
+
+Component member groups
+-----------------------
+
+Component member groups (CMGs) are groups of component members by type. We
+currently have the following groups:
+
+- databases
+- templates
+- static assets
+- hooks
+- plugins
+- WSGI middleware
+- route handlers
+- command line argument handlers
+
+Each component may have one or more members that belong to one of the CMGs. The
+CMGs are also known as component member type. 
+
+All members from all components that belong to the same group comprise that
+group. Although this may sound simple enough, it is worth keeping in mind that
+it *is* possible to step on other component's assets/templates/databases if you
+are not careful, as members are not namespaced. For example, if component A has
+a template named 'foo', and component B also has that template, the last
+component's template will always take precendece and override any previously
+found templates of the same name. Also, keep in mind that overriding templates
+in this way *intentionally* is also a bad idea, as the override order is an
+implementation detail that may change in future without notice.
+
+Component configuration file
+----------------------------
+
+Each component has a configuration file. This confugration file is in .ini
+format. By default, the configuration files will be looked up in 'config.ini'
+inside the package. A different location can be specified in the package's
+``__init__`` module by adding a :py:data:`CONFIG` constant. For example::
+
+    # mycomponent/__init__.py
+    CONFIG = 'configs/mycomponent.ini'
+
+This path is relative to the package directory and follows the same rules as
+the paths declared elsewhere in the configuration file itself: cannot contain
+double-dots, cannot be an absolute path, and so on.
+
+The component member exports are specified in the ``[exports]`` section of the
+configuration file.
+
+In the following subsections, we will repeat the section header every time, but
+do keep in mind that all exports go under a single ``[exports]`` heading.
+
+.. note::
+    The exact format used in the configuration files is covered `in confloader
+    module documentation
+    <http://confloader.readthedocs.org/en/latest/writing_ini.html>`_.
+
+Object and path references
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When specifying relative paths in the configuration options, you cannot use
+double-dot notation anywhere. For example, ``static_dir = ../foo`` is not
+allowed. The path resolver code will automatically strip out the ``../`` part.
```

Comment:
This should clarify that the paths used are relative to the component directory and not the config's directory